src/main/java/com/example/demo/repository/Tournament.java

```diff
@@ -0,0 +1,17 @@
+package com.example.demo.repository;
+
+import lombok.Data;
+import org.springframework.stereotype.Component;
+
+@Component
+@Data
+public class Tournament {
+    BaseTeamRepository teams;
+
+    Tournament(TeamRepository teamRepository){
```

Comment:
Good that you used injection here. To make it more flexible please use interface here(BaseTeamRepository). This way you can easily change implementation you wanna use for TeamRepository without changing code of this class